genienlp/models/transformer_seq2seq.py

```diff
@@ -44,25 +45,39 @@ def __init__(self, config=None, *inputs, args, tasks, vocab_sets, save_directory
         super().__init__(config)
         self.args = args
         args.dimension = config.d_model
+        self._is_bart_large = self.args.pretrained_model == 'facebook/bart-large'
+        self._is_mbart = 'mbart' in self.args.pretrained_model
+        
         if save_directory is not None:
             self.model = AutoModelForSeq2SeqLM.from_config(config)
         else:
             self.model = AutoModelForSeq2SeqLM.from_pretrained(self.args.pretrained_model,
                                                                cache_dir=self.args.embeddings)
+            
         self.numericalizer = TransformerNumericalizer(self.args.pretrained_model, max_generative_vocab=None,
                                                       preprocess_special_tokens=args.preprocess_special_tokens)
+
         self.init_vocab_from_data(vocab_sets, tasks, save_directory)
         self.model.resize_token_embeddings(self.numericalizer.num_tokens)
 
-        self._is_bart_large = self.args.pretrained_model == 'facebook/bart-large'
         if args.dropper_ratio > 0:
             self.dropper = LossDropper(dropc=args.dropper_ratio, min_count=args.dropper_min_count)
         else:
             self.dropper = None
-
+            
+            
     def add_new_vocab_from_data(self, tasks, resize_decoder=False):
         super().add_new_vocab_from_data(tasks, resize_decoder)
         self.model.resize_token_embeddings(self.numericalizer.num_tokens)
+    
+    
+    def set_decoder_start_token_id(self, lang):
+        if self._is_mbart:
+            # mBART, in contrast to MT5 or XLM-R, needs language id
+            # For now we only support single language training and evaluation with mbart models
+            lang_id = get_mbart_lang(lang)
+            self.model.config.decoder_start_token_id = self.numericalizer._tokenizer.lang_code_to_id[lang_id]
```

Comment:
Why is changing the model config necessary?

Reply:
It checks within the `decoder_start_token_id` method if model is mbart.
For TransformerSeq2Seq models it's always a _pass_.

Reply:
This is how huggingface training code accomplishes this. I think mainly because in generation forward pass, if `decoder_start_token_id` is not provided it will be determined inside `_get_decoder_start_token_id` in `generation_utils.py` from the config. 
It may have been used in other parts too. This being said, I think this code block can be removed (before our generation) but need to test it first before removing it. 

```        decoder_start_token_id = None
        if self._is_mbart:
            decoder_start_token_id = self.model.config.decoder_start_token_id
````